src/Adapter/Order/Repository/OrderRepository.php

```diff
@@ -71,4 +92,43 @@ public function get(OrderId $orderId): Order
 
         return $order;
     }
+
+    /**
+     * Get Order by cartId
+     *
+     * @param CartId $cartId
+     *
+     * @return Order|null
+     *
+     * @throws CoreException
+     */
+    public function findByCartId(CartId $cartId): ?Order
```

Comment:
```suggestion
    public function getByCartId(CartId $cartId): Order
```
To be more consistent with the usual `get` method in the object model repositories Also, instead of returning null maybe you should throw an `OrderNotFoundException`, also to be more consistent with the `get` behaviour

Reply:
Either that or you could implement a `public function getOrderIdByCartId(CartId $cartId): ?OrderId` method (this one could have a nullable return), which would match more with the usage you have (check if an order is associated to the cart) and it prevents fetching the whole `Order` object which you don't need in your delete handlers